includes/formatting.php

```diff
@@ -501,25 +501,21 @@ function give_format_amount( $amount, $args = array() ) {
  * @return string  formatted amount number with large number names.
  */
 function give_human_format_large_amount( $amount, $args = array() ) {
-	// Sanitize amount.
-	$sanitize_amount = give_maybe_sanitize_amount( $amount );
-
-	// Bailout.
-	if ( ! floatval( $sanitize_amount ) ) {
-		return '0';
-	};
 
 	// Set default currency;
 	if ( empty( $args['currency'] ) ) {
 		$args['currency'] = give_get_currency();
 	}
 
 	// Get thousand separator.
-	$thousands_sep = give_get_price_thousand_separator();
+	$thousands_sep = give_get_price_thousand_separator( $args['currency'] );
 
 	// Explode amount to calculate name of large numbers.
 	$amount_array = explode( $thousands_sep, $amount );
 
+	// Unformat currency for calculation purpose.
+	$unformat_amount = implode( '', $amount_array );
```

Comment:
@mehul0810 sanitize amount instead of unformatting because decimal separator can be `,`